File: old/internal/file/page_test.go
package file_test

import (
	"bytes"
	"testing"
	"time"

	"kura/internal/file"
)

func TestPage(t *testing.T) {
	// Arrange
	blockSize := int64(1024)
	page := file.NewPage(blockSize)

	// Act/Assert
	// Test SetInt and GetInt
	offsetInt := 0
	expectedInt := int32(123456)
	page.SetInt(offsetInt, expectedInt)
	if result := page.GetInt(offsetInt); result != expectedInt {
		t.Errorf("GetInt() = %d; want %d", result, expectedInt)
	}

	// Test SetShort and GetShort
	offsetShort := 8
	expectedShort := int16(12345)
	page.SetShort(offsetShort, expectedShort)
	if result := page.GetShort(offsetShort); result != expectedShort {
		t.Errorf("GetShort() = %d; want %d", result, expectedShort)
	}

	// Test SetBool and GetBool
	offsetBool := 16
	expectedBool := true
	page.SetBool(offsetBool, expectedBool)
	if result := page.GetBool(offsetBool); result != expectedBool {
		t.Errorf("GetBool() = %v; want %v", result, expectedBool)
	}

	// Test SetDate and GetDate
	offsetDate := 20
	expectedDate := time.Now().Truncate(time.Second) // Truncate to remove extra precision
	page.SetDate(offsetDate, expectedDate)
	if result := page.GetDate(offsetDate); !result.Equal(expectedDate) {
		t.Errorf("GetDate() = %v; want %v", result, expectedDate)
	}

	// Test SetBytes and GetBytes
	offsetBytes := 40
	expectedBytes := []byte{10, 20, 30, 40, 50}
	page.SetBytes(offsetBytes, expectedBytes)
	if result := page.GetBytes(offsetBytes); !bytes.Equal(result, expectedBytes) {
		t.Errorf("GetBytes() = %v; want %v", result, expectedBytes)
	}

	// Test SetString and GetString
	offsetString := 60
	expectedString := "hello world"
	page.SetString(offsetString, expectedString)
	if result := page.GetString(offsetString); result != expectedString {
		t.Errorf("GetString() = %q; want %q", result, expectedString)
	}
}

func TestMaxLength(t *testing.T) {
	strLen := 10
	expectedMaxLength := strLen + 1 // +1 for the null terminator
	if result := file.MaxLength(strLen); result != expectedMaxLength {
		t.Errorf("MaxLength(%d) = %d; want %d", strLen, result, expectedMaxLength)
	}
}

File: app/src/test/kotlin/com/kura/file/PageTest.kt
package com.kura.file

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test
import java.time.Instant

class PageTest {
    @Test
    fun `test page operations`() {
        // Arrange
        val blockSize = 1024
        val page = Page(blockSize)

        // Test Int operations
        val offsetInt = 0
        val expectedInt = 123456
        page.setInt(offsetInt, expectedInt)
        assertEquals(expectedInt, page.getInt(offsetInt), "Int value should match")

        // // Test Short operations
        // val offsetShort = 8
        // val expectedShort: Short = 12345
        // page.setShort(offsetShort, expectedShort)
        // assertEquals(expectedShort, page.getShort(offsetShort), "Short value should match")

        // // Test Boolean operations
        // val offsetBool = 16
        // val expectedBool = true
        // page.setBool(offsetBool, expectedBool)
        // assertEquals(expectedBool, page.getBool(offsetBool), "Boolean value should match")

        // // Test Date operations
        // val offsetDate = 20
        // val expectedDate = Instant.now().truncatedTo(java.time.temporal.ChronoUnit.SECONDS)
        // page.setDate(offsetDate, expectedDate)
        // assertEquals(expectedDate, page.getDate(offsetDate), "Date value should match")

        // Test Bytes operations
        val offsetBytes = 40
        val expectedBytes = byteArrayOf(10, 20, 30, 40, 50)
        page.setBytes(offsetBytes, expectedBytes)
        assertArrayEquals(expectedBytes, page.getBytes(offsetBytes), "Byte array should match")

        // Test String operations
        val offsetString = 60
        val expectedString = "hello world"
        page.setString(offsetString, expectedString)
        assertEquals(expectedString, page.getString(offsetString), "String value should match")
    }

    @Test
    fun `test maxLength calculation`() {
        val strLen = 10
        val expectedMaxLength = strLen + 4 // +4 is int size
        assertEquals(expectedMaxLength, Page.maxLength(strLen), "MaxLength should include null terminator")
    }

    @Test
    fun `test page creation from byte array`() {
        // Arrange
        val bytes = ByteArray(100)
        val value = 12345
        val offset = 0

        // Act
        val page = Page(bytes)
        page.setInt(offset, value)

        // Assert
        assertEquals(value, page.getInt(offset), "Value should be readable after writing to byte array-based page")
    }
}